github/github.go

```diff
@@ -336,7 +336,11 @@ func (c *Client) Do(req *http.Request, v interface{}) (*Response, error) {
 		return nil, err
 	}
 
-	defer resp.Body.Close()
+	defer func() {
+		// Drain and close the body to let the Transport reuse the connection
+		io.Copy(ioutil.Discard, resp.Body)
```

Comment:
Good question. Just tried and it turns out it's zero!

Maybe explicitly hitting `io.EOF` flips a flag somewhere, this looks like something the runtime could improve on.